Exiled.Events/Patches/Generic/RoomListAdd.cs

```diff
@@ -0,0 +1,59 @@
+// -----------------------------------------------------------------------
+// <copyright file="RoomListAdd.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Generic
+{
+#pragma warning disable SA1313
+    using System.Collections.Generic;
+    using System.Reflection.Emit;
+
+    using API.Features;
+
+    using HarmonyLib;
+    using MapGeneration;
+
+    using NorthwoodLib.Pools;
+    using UnityEngine;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    /// Patches <see cref="RoomIdentifier.Awake"/>.
+    /// </summary>
+    [HarmonyPatch(typeof(RoomIdentifier), nameof(RoomIdentifier.Awake))]
+    internal class RoomListAdd
+    {
+        private static void Prefix(RoomIdentifier __instance)
+        {
+            Room.RoomIdentifierToRoom.Add(__instance, Room.CreateComponent(__instance.gameObject));
+        }
+
+        // idk, it throws me real time error
```

Comment:
If the transpiler doesn't work, a prefix is fine for now, don't delete the comments.